Python/parking_lot.c

```diff
@@ -118,10 +118,20 @@ _PySemaphore_PlatformWait(_PySemaphore *sema, _PyTime_t timeout)
     if (timeout >= 0) {
         struct timespec ts;
 
+#if defined(CLOCK_MONOTONIC) && defined(HAVE_SEM_CLOCKWAIT)
+        _PyTime_t deadline = _PyTime_Add(_PyTime_GetMonotonicClock(), timeout);
+
+        _PyTime_AsTimespec_clamp(deadline, &ts);
+
+        err = sem_clockwait(&sema->platform_sem, CLOCK_MONOTONIC, &ts);
+#else
+        // does not support CLOCK_MONOTONIC, use system clock
```

Comment:
I'm not sure that this comment is useful, but if you want to keep it, I suggest:

```suggestion
        // Use system clock if sem_clockwait() is not available
```